src/public/modules/forms/admin/controllers/edit-fields-modal.client.controller.js

```diff
@@ -30,6 +30,26 @@ angular
     EditFieldsModalController,
   ])
 
+app.directive('validateEmailDomainFromText', () => {
+   return {
+      require: 'ngModel',
+      link: (scope, elem, attr, ngModel) => {
+        ngModel.$validators.emailDomainFromTextValidator = (allowedEmailDomainsFromText) => {
+          if (allowedEmailDomainsFromText) {
+            const emailDomains = allowedEmailDomainsFromText.split('\n').map(s => s.trim()).filter(s => s)
+            return (
+              !emailDomains.length ||
```

Comment:
I think this can be simplified/made more readable:

```ts
ngModel.$validators.emailDomainFromTextValidator = (allowedEmailDomainsFromText) => {
    // Do we need the last `filter` chain? Is it to remove falsey values? Add comment if so.
    const emailDomains = allowedEmailDomainsFromText.split('\n').map(s => s.trim()).filter(s => s)

    if (emailDomains.length === 0) return true
    const validDedupedDomainsSet = new Set(emailDomains.filter((s) => s.match(/@.+\..+/)))
    return validDedupedDomainsSet.size === emailDomains.length
}
```